lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1037,15 +1037,18 @@ SPIRVToLLVMDbgTran::ParseChecksum(StringRef Text) {
   // Example of "Text" variable:
   // "SomeInfo//__CSK_MD5:7bb56387968a9caa6e9e35fff94eaf7b:OtherInfo"
   Optional<DIFile::ChecksumInfo<StringRef>> CS;
-  auto KindPos = Text.find(SPIRVDebug::ChecksumKindPrefx);
-  if (KindPos != StringRef::npos) {
-    auto ColonPos = Text.find(":", KindPos);
-    KindPos += string("//__").size();
-    auto KindStr = Text.substr(KindPos, ColonPos - KindPos);
-    auto Checksum = Text.substr(ColonPos).ltrim(':');
-    if (auto Kind = DIFile::getChecksumKind(KindStr)) {
-      size_t ChecksumEndPos = Checksum.find_if_not(llvm::isHexDigit);
-      CS.emplace(Kind.getValue(), Checksum.substr(0, ChecksumEndPos));
+  if (Text.size() > SPIRVDebug::ChecksumKindPrefx.size() &&
```

Comment:
I think checking sizes is redundant. `find` should return `npos` in this case.

Reply:
If I understand StringRef correctly, "non-zero length and an empty data" can be possible only if StringRef object was constructed from nullptr. If I'm right we can just check here if Text.data() != nullptr

Reply:
@MrSidims Unfortunately, it doesn't work, because data is not quite empty, it contains '\0'

Reply:
@AlexeySotkin we need this because we are checking Text.front(), and if Text.size() is 0, calling Text.front() will cause an error. As I know, there isn't .at() method for StringRef, so we cant avoid this comparing. So I don't see another way to check it. Maybe you have another idea?

Reply:
Sorry, I still don't understand what we are fixing. Is it the case when `Text` contains only one character `\0`?
I think `find` on the next line should return `npos` in this case. Am I missing something?

Reply:
May we add a test for this patch? It might clarify the case that is fixed.

Reply:
@AlexeySotkin There is one moment in `StringRef::find` - `uint8_t Last = Start[N - 1];`
Start contains only '\0', so calling `N - 1` (where `N` is size of prefix `//__CSK_`) fails.